mne/simulation/tests/test_source.py

```diff
@@ -44,7 +45,8 @@ def test_generate_stc():
     tstep = 1e-3
 
     stc_data = np.ones((len(labels), n_times))
-    stc = generate_stc(fwd['src'], mylabels, stc_data, tmin, tstep)
+    with warnings.catch_warnings(record=True):
+        stc = generate_stc(fwd['src'], mylabels, stc_data, tmin, tstep)
```

Comment:
Looking at it, it's not as trivial to get this functionality as it was with `generate_stc` because of the forced use of `data_fun`. There used to be both `generate_stc` (which you could pass `data` to) and `generate_sparse_stc`, now there is only `simulate_sparse_stc`. Should there also be `simulate_stc` that takes a `data` parameter? I wasn't around for these discussions so I'm not sure.